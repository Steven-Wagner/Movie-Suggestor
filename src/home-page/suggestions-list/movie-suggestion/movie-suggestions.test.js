import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import MovieSuggestion from './movie-suggestion';

describe('<MovieSuggestions/>', () => {
    it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
        <Router>
            <MovieSuggestion movieData={{title: "Test Title", imdb_id: "imdb test", director: "Test Director", release_year: "2019", avg: '3'}}/>
        </Router>,
        div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(
            <Router>
                <MovieSuggestion movieData={{title: "Test Title", imdb_id: "imdb test", director: "Test Director", release_year: "2019", avg: '3'}}/>
            </Router>)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
});
