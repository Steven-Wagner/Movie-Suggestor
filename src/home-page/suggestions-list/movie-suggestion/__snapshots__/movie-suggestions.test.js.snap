// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieSuggestions/> renders the UI as expected 1`] = `
<section
  className="movie-suggestion remote-edge"
>
  <header
    className="movie-header"
  >
    <a
      className="IMDB-link"
      href="https://www.imdb.com/title/imdb test"
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2
        className="black-back movie-title"
      >
        Test Title
      </h2>
    </a>
    <a
      className="movie-poster-link"
      href="https://www.imdb.com/title/imdb test"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Test Title poster"
        className="movie-poster"
      />
    </a>
    <p
      className="movie-info"
    >
      Directed by: 
      Test Director
    </p>
    <p
      className="movie-info"
    >
      2019
    </p>
    <p
      className="star-rating"
    >
      ★★★☆☆
    </p>
  </header>
  <button
    className="suggestion-options-button remote-button"
    onClick={[Function]}
  >
    Not Interested
  </button>
  <button
    className="suggestion-options-button remote-button"
    onClick={[Function]}
  >
    Watched It
  </button>
</section>
`;
